### PR TITLE
Product routes

### PR DESCRIPTION
**Pull request by BurdockBH**

### CODE REVIEW

File: service/product/service.go [hunk 1]
```diff
@@ -56,3 +56,109 @@ func CreateProduct(w http.ResponseWriter, r *http.Request) {
...
+		response, _ := json.Marshal(viewmodels.BaseResponse{
+			StatusCode: statusCodes.FailedToDecodeRequestBody,
+			Message:    statusCodes.StatusCodes[statusCodes.FailedToDecodeRequestBody],
+		})
+		helper.BaseResponse(w, response, http.StatusBadRequest)
+		return
+	}
+	defer r.Body.Close()
+
+	_ = *helper.CheckToken(&w, r)
```
Comment (amina-hybird) [inline service/product/service.go:74]: what this function do?

File: service/product/service.go [hunk 2]
```diff
@@ -56,3 +56,109 @@ func CreateProduct(w http.ResponseWriter, r *http.Request) {
...
+
+	err = product.DeleteProduct(&p)
+	if err != nil {
+		log.Println("Failed to create product: ", err)
```
Comment (amina-hybird) [inline service/product/service.go:78]: fix log

File: service/product/service.go [hunk 3]
```diff
@@ -56,3 +56,109 @@ func CreateProduct(w http.ResponseWriter, r *http.Request) {
...
+		log.Println("Failed to decode request body: ", err)
+		response, _ := json.Marshal(viewmodels.BaseResponse{
+			StatusCode: statusCodes.FailedToDecodeRequestBody,
+			Message:    statusCodes.StatusCodes[statusCodes.FailedToDecodeRequestBody],
+		})
+		helper.BaseResponse(w, response, http.StatusBadRequest)
+		return
+	}
+	defer r.Body.Close()
+
```
Comment (amina-hybird) [inline service/product/service.go:108]: where is req validation

File: service/product/service.go [hunk 4]
```diff
@@ -56,3 +56,109 @@ func CreateProduct(w http.ResponseWriter, r *http.Request) {
...
```
Comment (amina-hybird) [inline service/product/service.go:143]: req validation

File: migrations/20230626154913_edit_product.sql [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+CREATE PROCEDURE `EditProduct`(
+    IN inProductId int,
+    IN inProductName VARCHAR(50),
+    IN inProductDescription VARCHAR(255),
+    IN inProductPrice FLOAT
+)
+BEGIN
+
+    IF (SELECT id FROM products WHERE id = inProductId) IS NULL THEN
+        SELECT 0;
```
Comment (amina-hybird) [inline migrations/20230626154913_edit_product.sql:12]: you can put everything in one IF

IF EXISTS(..) THEN

SELECT 1
END IF;
In code if select from db is != 1 then it is not updated

File: migrations/20230626155016_get_products.sql [hunk 6]
```diff
@@ -0,0 +1,57 @@
+-- +goose Up
+-- +goose StatementBegin
+CREATE PROCEDURE `GetProducts`(
+    IN product_name VARCHAR(255),
```
Comment (amina-hybird) [inline migrations/20230626155016_get_products.sql:4]: cliend will probably send to you venueID, so you don't need check name or address

File: service/product/service.go (hunk 1 above)
Comment (BurdockBH) [reply service/product/service.go:74]: Authorizes token. It returns claims, in some functions I use those claims, in other, where not needed, it just checks. It handles errors and response inside.

**Pull request merged**